go/vt/vtgate/vcursor_impl.go

```diff
@@ -753,6 +753,11 @@ func (vc *vcursorImpl) WarnUnshardedOnly(format string, params ...interface{}) {
 	}
 }
 
+// ForeignKey implements the VCursor interface
+func (vc *vcursorImpl) ForeignKeyMode() string {
+	return *foreignKeyMode
```

Comment:
consider `return strings.ToLower(*foreignKeyMode)` so that clients can immediately use as map key.